base/tps/src/org/dogtagpki/server/tps/TPSTokendb.java

```diff
@@ -243,25 +243,76 @@ public void tdbAddCertificatesForCUID(String cuid, ArrayList<TPSCertRecord> cert
 
         logger.debug(method + " found token " + cuid);
         logger.debug(method + " number of certs to update:" + certs.size());
+
+	// Keep track of which certs made it to the database and which didn't,
+        // in case of failure
+         class CnIssuerPair {
+             public final String cn;
+             public final String issuer;
+
+             public CnIssuerPair(String _cn, String _issuer) {
+                 cn = _cn;
+                 issuer = _issuer;
+             }
+
+             public String toString() {
```

Comment:
are we certain that these two values will not be null?

Reply:
The values come from data from the tokendb, so the values shoudl be good, but even if they are null, it will just show up in the log message as null, which will also be a clue to the admin that something is wrong.